AGENTS.md

```diff
@@ -0,0 +1,258 @@
+
+```
+insights-behavioral-spec/
+├── config/              # Configuration files for various services (TOML, YAML, ENV)
+├── docs/                # Documentation (BDD intro, feature lists, tools, etc.)
+├── features/            # BDD test specifications and implementations
+│   ├── ACM/             # Advanced Cluster Management (multi-cluster) tests
+│   ├── ccx-notification-service/ # Notification service tests
+│   ├── ccx-notification-writer/  # Notification writer tests
+│   ├── ccx-upgrades-data-eng/    # Data engineering service tests
+│   ├── ccx-upgrades-inference/   # Inference service tests
+│   ├── DVO_Recommendations/      # DVO recommendation tests
+│   ├── dvo-extractor/            # DVO extractor tests
+│   ├── dvo-writer/               # DVO writer tests
+│   ├── Insights_Advisor/         # Insights Advisor tests
+│   ├── insights-content-service/ # Content service tests
+│   ├── insights-content-template-renderer/  # Template renderer tests
+│   ├── insights-results-aggregator/         # Aggregator tests
+│   ├── insights-results-aggregator-cleaner/ # Cleaner tests
+│   ├── insights-results-aggregator-exporter/# Exporter tests
+│   ├── insights-results-aggregator-mock/    # Mock service tests
+│   ├── OCM/                      # OpenShift Cluster Manager tests
+│   ├── OCP_WebConsole/           # OCP Web Console tests
+│   ├── parquet-factory/          # Parquet factory tests
+│   ├── SHA_Extractor/            # SHA extractor tests
+│   ├── smart-proxy/              # Smart proxy tests
+│   ├── src/                      # Shared Python utilities
+│   ├── steps/                    # Behave step implementations
+│   ├── environment.py            # Behave environment setup (actions to run before and after testing)
+│   └── environment_test.py       # Environment test utilities
+├── logs/                # Test execution logs organized by service
+├── mocks/               # Mock services for testing (content-service, prometheus, etc.)
+├── setup/               # Database setup scripts (SQL files)
+├── test_data/           # Test data files (JSON, compressed files)
+├── test_list/           # Lists of test scenarios per service
+├── tools/               # Utility scripts (scenario list generator, code style checker)
+├── *_tests.sh           # Service-specific test execution scripts
+├── docker-compose.yml   # Docker Compose configuration for test infrastructure
+├── Makefile             # Build and test targets
+├── requirements.txt     # Python dependencies
+└── run_tests.sh         # Main test runner script
+```
+
+### Key Directories
+
+- **`features/`**: Contains all BDD feature files (`.feature`) and their step implementations. Each service has its own subdirectory. You should use different files files for different features, for example if you are testing an integration with a service, dedicate one file for that specific purpose. This helps organizing the scenarios and makes it easier to understand the purpose of each test.
```

Comment:
Just a question: why using too long lines instead of breaking it into multiple shorter ones?